x/shield/keeper/purchase.go

```diff
@@ -165,12 +194,22 @@ func (k Keeper) RemoveExpiredPurchases(ctx sdk.Context) {
 			for i := 0; i < len(purchaseList.Entries); {
 				entry := purchaseList.Entries[i]
 				// DeletionTime = ProtectionEndTime - ProtectionPeriod + ClaimPeriod + VotingPeriod
-				// If DeletionTime > Now, skip.
+				// If DeletionTime > currentBlockTime, skip.
 				if entry.ProtectionEndTime.Add(deletionPeriod).After(ctx.BlockTime()) {
 					i++
```

Comment:
How come you are only incrementing i here?

Reply:
Because if DeletionTime <= currentBlockTime the entry will be removed, so we don't need to add one in that case.